### DISCUSSION TITLE
Enhancement: Azure OpenAI On Your Data API support

### DISCUSSION THREAD

**Discussion by shota1998** [danny-avila/LibreChat#2724] [Feature Requests & Suggestions]
### What features would you like to see added?

Users can use Azure AI Search without a plugin.

### More details

Official references:
- [Data source - Azure AI Search](https://learn.microsoft.com/en-us/azure/ai-services/openai/references/azure-search?tabs=python#examples)
- [Azure OpenAI On Your Data API Reference](https://learn.microsoft.com/en-us/azure/ai-services/openai/references/on-your-data?tabs=rest#examples)

Current situation:
- Users need to select Azure AI Search from a plugin when using it.
- Every time the index is switched, the plugin needs to be reinstalled.

Request:
- Developers can configure the Azure AI Search and index to be used in the librechat.yaml file.

Advantages:
- By specifying multiple configurations, it is possible to switch between multiple indexes.
- Users can use Azure AI Search when they are not allowed to use plugins.
- Developer can force users to use Azure AI Search.

### Which components are impacted by your request?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You can now set `modelSpecs` which have plugins "pre-installed."

I know what you mean by the index being provided as a credential, and I think we can definitely improve that soon by allowing certain tool parameters to be pre-configured.
